miio/integrations/vacuum/dreame/dreamevacuum_miot.py

```diff
@@ -190,12 +256,115 @@ def volume(self) -> str:
+    def cleaning_time(self) -> str:
+        return self.data["cleaning_time"]
+
+    @property
+    def cleaning_area(self) -> str:
+        return self.data["cleaning_area"]
+
+    @property
+    def first_clean_time(self) -> str:
+        return self.data["first_clean_time"]
+
+    @property
+    def total_clean_time(self) -> str:
+        return self.data["total_clean_time"]
+
+    @property
+    def total_clean_times(self) -> str:
+        return self.data["total_clean_times"]
+
+    @property
+    def total_clean_area(self) -> str:
+        return self.data["total_clean_area"]
+
+    @property
+    def cleaning_mode(self):
+        cleaning_mode = self.data["cleaning_mode"]
+        if self.model == DREAME_1C:
+            try:
+                return CleaningModeDreame1C(cleaning_mode)
+            except ValueError:
+                _LOGGER.error(f"Unknown CleaningMode ({cleaning_mode})")
+                return CleaningModeDreame1C.Unknown
+        elif self.model in [DREAME_D9, DREAME_F9]:
+            try:
+                return CleaningModeDreameF9(cleaning_mode)
+            except ValueError:
+                _LOGGER.error(f"Unknown CleaningMode ({cleaning_mode})")
+                return CleaningModeDreameF9.Unknown
 
-    mapping = _MAPPING
-    _supported_models = ["dreame.vacuum.mc1808"]
+    @property
+    def life_sieve(self) -> Optional[str]:
+        try:
+            return self.data["life_sieve"]
+        except KeyError:
+            return None
+
+    @property
+    def life_brush_side(self) -> Optional[str]:
+        try:
+            return self.data["life_brush_side"]
+        except KeyError:
+            return None
+
+    @property
+    def life_brush_main(self) -> Optional[str]:
+        try:
+            return self.data["life_brush_main"]
+        except KeyError:
+            return None
+
+    @property
+    def water_flow(self) -> Optional[WaterFlow]:
+        try:
+            water_flow = self.data["water_flow"]
+        except KeyError:
+            return None
+        try:
+            return WaterFlow(water_flow)
+        except ValueError:
+            _LOGGER.error("Unknown WaterFlow (%s)", self.data["water_flow"])
+            return WaterFlow.Unknown
+
+    @property
+    def is_water_box_carriage_attached(self) -> Optional[bool]:
+        """Return True if water box carriage (mop) is installed, None if sensor not
+        present."""
+        if "water_box_carriage_status" in self.data:
+            return self.data["water_box_carriage_status"] == 1
+        return None
+
+
+class DreameVacuumBase(MiotDevice):
+    _supported_models = [
+        DREAME_1C,
+        DREAME_D9,
+        DREAME_F9,
+    ]
+
+    mapping = MIOT_MAPPING[DREAME_1C]
+
+    def __init__(self, *args, **kwargs) -> None:
+        model = kwargs.get("model")
+        if model:
+            mapping = MIOT_MAPPING.get(model)
+        if mapping:
+            self.mapping = mapping
+        super().__init__(*args, **kwargs)
```

Comment:
This is not needed, `MiotDevice` class takes care of using the correct mapping for methods since that recent PR.

Reply:
@rytilahti  yep. It is not 100% ready yet. It was changed after merge. I will write a message when this pr will be ready